osmoutils/accum/accum.go

```diff
@@ -151,13 +214,45 @@ func (accum AccumulatorObject) RemoveFromPosition(name string, numSharesToRemove
 		return err
 	}
 
-	createNewPosition(accum, name, oldNumShares.Sub(numSharesToRemove), unclaimedRewards, position.Options)
+	createNewPosition(accum, accumulatorUpdate, name, oldNumShares.Sub(numSharesToRemove), unclaimedRewards, position.Options)
 
 	return nil
 }
 
-// GetPositionSize returns the number of shares the position corresponding to postion's name
-// or an error if no position exists.
+// UpdatePosition updates the position with the given name by adding or removing
+// the given number of shares. If numShares is positive, it is equivalent to calling
+// AddToPosition. If numShares is negative, it is equivalent to calling RemoveFromPosition.
+// Also, it moves up the position's accumulator value to the current accum value.
+// Fails with error if numShares is zero. Returns nil on success.
+func (accum AccumulatorObject) UpdatePosition(name string, numShares sdk.Dec) error {
+	return accum.updatePosition(name, numShares, accum.value)
+}
+
+// UpdatePositionCustomAcc updates the position with the given name by adding or removing
+// the given number of shares. If numShares is positive, it is equivalent to calling
+// AddToPositionCustomAcc. If numShares is negative, it is equivalent to calling RemoveFromPositionCustomAcc.
+// Fails with error if numShares is zero. Returns nil on success.
+// It also resets the position's accumulator value to the given customAccumulatorValue.
+// All custom accumulator values must be non-negative. They must also be a superset of the
+// old accumulator value associated with the position.
+func (accum AccumulatorObject) UpdatePositionCustomAcc(name string, numShares sdk.Dec, customAccumulatorValue sdk.DecCoins) error {
+	return accum.updatePosition(name, numShares, customAccumulatorValue)
+}
+
+func (accum AccumulatorObject) updatePosition(name string, numShares sdk.Dec, positionAccumulatorUpdate sdk.DecCoins) error {
```

Comment:
why not directly expose `updatePosition` if this is the case?

Reply:
Good point. Refactored APIs here to reduce boilerplate: [54042ac](https://github.com/osmosis-labs/osmosis/pull/3903/commits/54042ac07577e4ef8b043e5ad3d7d3181e0770a8)